### ISSUE TITLE
[FEATURE] Implement DELETE API for the workflow

### ISSUE THREAD

**Issue by owaiskazi19** [opensearch-project/flow-framework#23]
### Is your feature request related to a problem?
This API would take care of deleting the workflow setup which was created before. This API should also handle if at any point workflow setup gets blocked or didn't complete successfully, associated resources created until then would be deleted from OpenSearch. 

### What solution would you like?
`DELETE _plugins/ai-flow?id=request-id`

This API would delete specific workflow setup. We would need another API to delete multiple resources

```
DELETE _plugins/ai-flow
{
    workflowIds: [abc, xyz, 123]
}
```

### Do you have any additional context?
Before deleting the resources, we need to check if there's any dependency on multiple resources and if the resource is safe to delete. This API should help to free up the space from OpenSearch created by the plugin.

**Comment by owaiskazi19** [comment]
This API is dependent on the META issue: https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/32

**Comment by dbwiddis** [comment]
Fixed in #294
